Objective-C/CBLDatabase.mm

```diff
@@ -178,12 +178,16 @@ - (NSString*) description {
 
 
 - (NSString*) path {
-    return _c4db != nullptr ? sliceResult2FilesystemPath(c4db_getPath(_c4db)) : nil;
+    @synchronized (self) {
```

Comment:
Other classes can lock (@synchronized) on CBLDatabase instance.  This could cause the dead-lock or other threading problems.

In most of the cases, you would like to protect c4Database instance instead of CBLDatabase. This is main reason that I asked this question.